contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+                _liquidity
+            );
+
+        require(_wethAmount == 0, "Non zero wethAmount");
+        underlyingAssets = _oethbAmount;
+        emit UnderlyingAssetsUpdated(underlyingAssets);
+    }
+
+    /**
+     * @notice Deposit WETH to the strategy contract. This function does not add liquidity to the
+     *         underlying Aerodrome pool.
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance > 0) {
+            _deposit(WETH, _wethBalance);
+        }
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     *         send to the `_recipient`.
+     * @param _recipient  Address to which the asset should be sent
+     * @param _asset      WETH address
+     * @param _amount     Amount of WETH to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_asset == WETH, "Unsupported asset");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance < _amount) {
+            uint256 _additionalWethRequired = _amount - _wethBalance;
+            (uint256 _wethInThePool, ) = getPositionPrincipal();
+
+            if (_wethInThePool < _additionalWethRequired) {
+                revert NotEnoughWethLiquidity(
+                    _wethInThePool,
+                    _additionalWethRequired
+                );
+            }
+
+            uint256 shareOfWethToRemove = Math.min(
+                _additionalWethRequired.divPrecisely(_wethInThePool) + 1,
+                1e18
+            );
+            _removeLiquidity(shareOfWethToRemove);
+        }
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        _withdraw(_recipient, _amount);
+    }
+
+    /**
+     * @notice Withdraw WETH and sends it to the Vault.
+     */
+    function withdrawAll() external override onlyVault nonReentrant {
+        if (tokenId > 0) {
+            _removeLiquidity(1e18);
+        }
+
+        uint256 _balance = IERC20(WETH).balanceOf(address(this));
+        if (_balance > 0) {
+            _withdraw(vaultAddress, _balance);
+        }
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+    }
+
+    function _withdraw(address _recipient, uint256 _amount) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+    /**
+     * @dev Collect the AERO token from the gauge
+     */
+    function _collectRewardTokens() internal override {
+        if (tokenId > 0) {
```

Comment:
This is assuming that the token is deposited into the gauge. In the new code that can be non-true. I think this method will revert if it is not?

Reply:
Yes great point the call [would revert here](https://github.com/aerodrome-finance/slipstream/blob/main/contracts/gauge/CLGauge.sol#L164). The fix for it in this commit: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/b30756ad7e0da7d724372ebee144ca4b9fe5ea10